sarif/sarif.go

```diff
@@ -225,7 +210,10 @@ func (s *SarifTransformer) transformToOCSF(toolName string, res *sarif.Result) (
 
 	// Location
 	var fixAvailable = ptr.Ptr(len(res.Fixes) > 0)
-
+	ds, err := protojson.Marshal(s.datasource)
```

Comment:
I'm not sure I understand this bit 🤔 

You want to allow parsers to map the datasource as they know they can? Basically duplicating what `mapDataSource` was doing?

Also, I'd expect `ToOCSF` to take in input the data source since it differ for each finding (potentially) - right now it looks like I'd have to create a new Transformer every time

Let me know if I'm misunderstanding!

Reply:
you are right, datasource contains both scan metadata and finding metadata, fixed, addressed and i think ready